lib/iris/config.py

```diff
@@ -154,3 +168,267 @@ def get_dir_option(section, option, default=None):
+            graph in parallel. Defaults to 1 (that is, processed serially).
+
+            .. note::
+                The value for `num_workers` cannot be set to greater than the
+                number of CPUs available on the host system. If such a value is
+                requested, `num_workers` is automatically set to 1 less than
+                the number of CPUs available on the host system.
+
+            .. note::
+                Only the 'threaded' and 'multiprocessing' schedulers support
+                the `num_workers` kwarg. If it is specified with the `async` or
+                `distributed` scheduler, the kwarg is ignored:
+
+                * The 'async' scheduler runs serially so will only use a single
+                worker.
+                * The number of workers for the 'distributed' scheduler must be
+                defined when setting up the distributed scheduler. For more
+                information on setting up distributed schedulers, see
+                https://distributed.readthedocs.io/en/latest/index.html.
+
+        Example usages:
+
+        * Specify that we want to load a cube with dask parallel processing
+        using multiprocessing with six worker processes::
+
+            iris.options.parallel(scheduler='multiprocessing', num_workers=6)
+            iris.load('my_dataset.nc')
+
+        * Specify, with a context manager, that we want to load a cube with
+        dask parallel processing using four worker threads::
+
+            with iris.options.parallel(scheduler='threaded', num_workers=4):
+                iris.load('my_dataset.nc')
+
+        * Run dask parallel processing using a distributed scheduler that has
+        been set up at the IP address and port at ``192.168.0.219:8786``::
+
+            iris.options.parallel(scheduler='192.168.0.219:8786')
+
+        """
+        # Set `__dict__` keys first.
+        self.__dict__['_scheduler'] = scheduler
+        self.__dict__['scheduler'] = None
+        self.__dict__['num_workers'] = None
+        self.__dict__['dask_scheduler'] = None
+
+        # Set `__dict__` values for each kwarg.
+        setattr(self, 'scheduler', scheduler)
+        setattr(self, 'num_workers', num_workers)
+        setattr(self, 'dask_scheduler', self.get('scheduler'))
+
+        # Activate the specified dask options.
+        self._set_dask_options()
+
+    def __repr__(self):
+        msg = 'Dask parallel options: {}.'
+
+        # Automatically populate with all currently accepted kwargs.
+        options = ['{}={}'.format(k, v)
+                   for k, v in six.iteritems(self.__dict__)
+                   if not k.startswith('_')]
+        joined = ', '.join(options)
+        return msg.format(joined)
+
+    def __setattr__(self, name, value):
+        if value is None:
+            value = self._defaults_dict[name]['default']
+        attr_setter = getattr(self, 'set_{}'.format(name))
```

Comment:
This is pretty ugly, IMO. I like the way Matplotlib handles this. In addition to storing default values, they also store a validator/converter function for each parameter (See [matplotlib/rcsetup.py](https://github.com/matplotlib/matplotlib/blob/4ba8903b83e41e4f83cc459ba0c3202dfd58c5d3/lib/matplotlib/rcsetup.py#L964))